src/main/java/com/rabbitmq/client/impl/recovery/TopologyRecoveryRetryLogic.java

```diff
@@ -165,14 +164,52 @@
                 } else if (consumer.getChannel() == channel) {
                     final RetryContext retryContext = new RetryContext(consumer, context.exception(), context.connection());
                     RECOVER_CONSUMER_QUEUE.call(retryContext);
-                    context.connection().recoverConsumer(consumer.getConsumerTag(), consumer, false);
+                    context.connection().recoverConsumer(consumer.getConsumerTag(), consumer);
```

Comment:
similar concept here as with the recoverQueue... we want this to throw an exception rather than deliver to the exceptionhandler and swallow the exception